Mathlib/Algebra/Order/Archimedean/Basic.lean

```diff
@@ -38,74 +40,102 @@ class Archimedean (α) [OrderedAddCommMonoid α] : Prop where
   such that `x ≤ n • y`. -/
   arch : ∀ (x : α) {y : α}, 0 < y → ∃ n : ℕ, x ≤ n • y
 
-instance OrderDual.archimedean [OrderedAddCommGroup α] [Archimedean α] : Archimedean αᵒᵈ :=
-  ⟨fun x y hy =>
-    let ⟨n, hn⟩ := Archimedean.arch (-ofDual x) (neg_pos.2 hy)
-    ⟨n, by rwa [neg_nsmul, neg_le_neg_iff] at hn⟩⟩
+section MulArchimedean
 
-variable {M : Type*}
+/-- An ordered commutative monoid is called `MulArchimedean` if for any two elements `x`, `y`
+such that `1 < y`, there exists a natural number `n` such that `x ≤ y ^ n`. -/
+class MulArchimedean (α) [OrderedCommMonoid α] : Prop where
+  /-- For any two elements `x`, `y` such that `1 < y`, there exists a natural number `n`
+  such that `x ≤ y ^ n`. -/
+  arch : ∀ (x : α) {y : α}, 1 < y → ∃ n : ℕ, x ≤ y ^ n
 
-theorem exists_lt_nsmul [OrderedAddCommMonoid M] [Archimedean M]
-    [CovariantClass M M (· + ·) (· < ·)] {a : M} (ha : 0 < a) (b : M) :
-    ∃ n : ℕ, b < n • a :=
-  let ⟨k, hk⟩ := Archimedean.arch b ha
-  ⟨k + 1, hk.trans_lt <| nsmul_lt_nsmul_left ha k.lt_succ_self⟩
+attribute [to_additive Archimedean] MulArchimedean
+end MulArchimedean
 
-section LinearOrderedAddCommGroup
+@[to_additive]
+instance OrderDual.mulArchimedean [OrderedCommGroup α] [MulArchimedean α] : MulArchimedean αᵒᵈ :=
+  ⟨fun x y hy =>
+    let ⟨n, hn⟩ := MulArchimedean.arch (ofDual x)⁻¹ (inv_lt_one_iff_one_lt.2 hy)
+    ⟨n, by rwa [inv_pow, inv_le_inv_iff] at hn⟩⟩
 
-variable [LinearOrderedAddCommGroup α] [Archimedean α]
+instance Additive.Archimedean [OrderedCommGroup α] [MulArchimedean α] : Archimedean (Additive α) :=
```

Comment:
```suggestion
instance Additive.instArchimedean [OrderedCommGroup α] [MulArchimedean α] : Archimedean (Additive α) :=
```

Reply:
I can rename them as such -- should the existing `OrderDual.archimedean` be renamed also?

Reply:
They should, yes